docs/pages/management/guides/joining-nodes-azure.mdx

```diff
@@ -0,0 +1,399 @@
+$ ROLE_NAME=$(echo $DEPLOY_OUTPUT | jq -r '.properites.outputs.roleName.value')
+```
+
+Next, create another file named `teleport-assign-identity.json` and copy the
+following into it:
+
+```json
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "metadata": {
+    "_generator": {
+      "name": "bicep",
+      "version": "0.5.6.12127",
+      "templateHash": "2227781763411200690"
+    }
+  },
+  "parameters": {
+    "identityName": {
+      "type": "string",
+      "defaultValue": "teleport-azure"
+    },
+    "principalId": {
+      "type": "string"
+    },
+    "roleName": {
+      "type": "string"
+    },
+    "vmNames": {
+      "type": "array"
+    }
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Authorization/roleAssignments",
+      "apiVersion": "2022-04-01",
+      "name": "[guid(resourceGroup().id)]",
+      "properties": {
+        "roleDefinitionId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', parameters('roleName'))]",
+        "principalId": "[parameters('principalId')]"
+      }
+    },
+    {
+      "apiVersion": "2018-06-01",
+      "type": "Microsoft.Compute/virtualMachines",
+      "name": "[parameters('vmNames')[copyIndex('vmcopy')]]",
+      "location": "[resourceGroup().location]",
+      "identity": {
+        "type": "userAssigned",
+        "userAssignedIdentities": {
+          "[resourceID('Microsoft.ManagedIdentity/userAssignedIdentities/',parameters('identityName'))]": {}
+        }
+      },
+      "copy": {
+        "name": "vmcopy",
+        "count": "[length(parameters('vmNames'))]"
+      }
+    }
+  ]
+}
+```
+
+Run the following command to assign your custom role to your managed identity
+and assign the identity to your virtual machines:
+
+```code
+$ az deployment group create \
+--resource-group <your-resource-group> \
+--template-file teleport-assign-identity.json \
+-- parameters principalId="$PRINCIPAL_ID" roleName="$ROLE_NAME" \
+vmNames='<list-of-vm-names>'
+```
+
+Use the value of `CLIENT_ID` in Step 3.
+</TabItem>
+</Tabs>
+
+## Step 2/4. Create the Azure Node joining token
+Under the hood, Nodes will prove that they are running in your Azure subscription
+by sending a signed attested data document and access token to the Teleport Auth
+Service. The Node's identity must match an allow rule configured in your Azure
+Node joining token.
+
+Create the following `token.yaml` with an `allow` rule specifying your Azure
+subscription and the resource group that your Node's identity must match.
+
+```yaml
+# token.yaml
+kind: token
+version: v2
+metadata:
+  # the token name is not a secret because instances must prove that they are
+  # running in your Azure subscription to use this token
+  name: azure-token
+  # set a long expiry time, the default for tokens is only 30 minutes
+  expires: "3000-01-01T00:00:00Z"
```

Comment:
@strideynet this is no longer necessary, right?

Reply:
@strideynet Should we remove it from master and only added when backporting to v12? Sounds like adding it everywhere just encourages "bad practices" that are not longer necessary.

Reply:
That sounds like a good plan !